File: samples/client/3_0_3_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/UnsetAddPropsSetter.java
package org.openapijsonschematools.client.schemas;

import org.checkerframework.checker.nullness.qual.Nullable;
import org.openapijsonschematools.client.exceptions.InvalidAdditionalPropertyException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;

import java.util.List;
import java.util.Map;
import java.util.Set;

public abstract class UnsetAddPropsSetter<T> {
    public abstract Map<String, @Nullable Object> getInstance();
    public abstract Set<String> getKnownKeys();
    public abstract T getBuilderAfterAdditionalProperty(Map<String, @Nullable Object> instance);
    public T additionalProperty(String key, Void value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }

    public T additionalProperty(String key, boolean value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }

    public T additionalProperty(String key, String value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }

    public T additionalProperty(String key, int value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }

    public T additionalProperty(String key, float value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }

    public T additionalProperty(String key, long value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }

    public T additionalProperty(String key, double value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }

    public T additionalProperty(String key, List<?> value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }

    public T additionalProperty(String key, Map<String, ?> value) throws InvalidAdditionalPropertyException {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true);
        var instance = getInstance();
        instance.put(key, value);
        return getBuilderAfterAdditionalProperty(instance);
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/UnsetAddPropsSetter.kt
package org.openapijsonschematools.client.schemas

import org.openapijsonschematools.client.exceptions.InvalidAdditionalPropertyException
import org.openapijsonschematools.client.schemas.validation.MapUtils

interface UnsetAddPropsSetter<T> {
    fun getInstance(): MutableMap<String, Any?>
    fun getKnownKeys(): Set<String>
    fun getBuilderAfterAdditionalProperty(instance: MutableMap<String, Any?>): T
    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: Void): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }

    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: Boolean): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }

    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: String): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }

    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: Int): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }

    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: Float): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }

    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: Long): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }

    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: Double): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }

    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: List<*>): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }

    @Throws(InvalidAdditionalPropertyException::class)
    fun additionalProperty(key: String, value: Map<String?, *>): T {
        MapUtils.throwIfKeyKnown(key, getKnownKeys(), true)
        val instance = getInstance()
        instance[key] = value
        return getBuilderAfterAdditionalProperty(instance)
    }
}